### PR TITLE
fix: reservation edit & cancel pages require authentication

### PR DESCRIPTION
**Pull request by vincit-matu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Doesn't show reservation edit or cancel-pages if the reservation doesn't belong to the logged in user

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Try to access the edit/cancel pages of a reservation when logged in as some other user than the one to whom the reservation belongs

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- TILA-3362

### CODE REVIEW

File: apps/ui/pages/reservations/[id]/cancel.tsx [hunk 1]
```diff
@@ -54,15 +61,17 @@ export async function getServerSideProps(ctx: GetServerSidePropsContext) {
...
-    return {
-      props: {
-        ...commonProps,
-        ...(await serverSideTranslations(locale ?? "fi")),
-        key: `${pk}-cancel-{locale}`,
-        reservation: reservation ?? null,
-        reasons,
-      },
-    };
+    if (reservation != null && reservation.user?.pk === user?.pk) {
```
Comment (joonatank) [inline apps/ui/pages/reservations/[id]/cancel.tsx:64]: just to make sure, check that one of the pk:s is not null so `user.pk === null` doesn't fall through

File: apps/ui/pages/reservations/[id]/cancel.tsx [hunk 2]
```diff
@@ -54,15 +61,17 @@ export async function getServerSideProps(ctx: GetServerSidePropsContext) {
     );
 
     // TODO check for null and return notFound
```
Comment (joonatank) [inline apps/ui/pages/reservations/[id]/cancel.tsx:63]: can remove

File: apps/ui/pages/reservations/[id]/cancel.tsx [hunk 3]
```diff
@@ -54,15 +61,17 @@ export async function getServerSideProps(ctx: GetServerSidePropsContext) {
...
-        reasons,
-      },
-    };
+    if (reservation != null && reservation.user?.pk === user?.pk) {
+      return {
+        props: {
+          ...commonProps,
+          ...(await serverSideTranslations(locale ?? "fi")),
+          key: `${pk}-cancel-{locale}`,
+          reservation: reservation ?? null,
```
Comment (joonatank) [inline apps/ui/pages/reservations/[id]/cancel.tsx:70]: can remove `??...`

**Review by joonatank — APPROVED** [review]
bit cleanup otherwise 👍

File: apps/ui/pages/reservations/[id]/cancel.tsx (hunk 3 above)
Comment (vincit-matu) [reply apps/ui/pages/reservations/[id]/cancel.tsx:70]: yeah,`?? null` seems to be a bit redundant, but `?? "fi"` seems to be necessary as a fallback

**Comment by joonatank** [comment]
Merged https://github.com/City-of-Helsinki/tilavarauspalvelu-ui/pull/1238, https://github.com/City-of-Helsinki/tilavarauspalvelu-ui/pull/1245, https://github.com/City-of-Helsinki/tilavarauspalvelu-ui/pull/1231, https://github.com/City-of-Helsinki/tilavarauspalvelu-ui/pull/1272,  https://github.com/City-of-Helsinki/tilavarauspalvelu-ui/pull/1333 into this.

**Pull request merged**